src/Features/Core/Portable/EmbeddedLanguages/RegularExpressions/LanguageServices/RegexPatternDetector.cs

```diff
@@ -283,6 +283,16 @@ public bool IsRegexPattern(SyntaxToken token, CancellationToken cancellationToke
                     }
                 }
             }
+            else if (syntaxFacts.IsImplicitObjectCreationExpression(invocationOrCreation))
+            {
+                var constructor = _semanticModel.GetSymbolInfo(invocationOrCreation, cancellationToken).GetAnySymbol();
```

Comment:
In the case of an object creation (`new Regex()`), there is a quick check on the type name before querying the semantic model.
Do you think it is useful to try to detect the type name here before going to the semantic model? For instance, we could try to get the type from the left side of the assignation (`Regex r = new("")`). If so, which pattern should we support?

Reply:
Unfortunately, I can't see a way that would help. Implicit object creation could show up anywhere (like `Foo(new(""))`. So we're going to need semantics no matter what.